app/controllers/sensors_controller.rb

```diff
@@ -1,7 +1,15 @@
 class SensorsController < ApplicationController
+  before_action :user_authenticate, only:[:index, :show]
+
   def index
+    @sensors = Sensor.where(user_id: params[:user_id])
```

Comment:
This will allow you to do eager loading for your sensors, and include their readings: http://apidock.com/rails/ActiveRecord/QueryMethods/includes